node/core/backing/src/lib.rs

```diff
@@ -0,0 +1,1720 @@
+		}
+
+		Ok(valid.0)
+	}
+
+	fn get_backed(&self) -> Vec<NewBackedCandidate> {
+		let proposed = self.table.proposed_candidates(&self.table_context);
+		let mut res = Vec::with_capacity(proposed.len());
+
+		for p in proposed.into_iter() {
+			let TableAttestedCandidate { candidate, validity_votes, .. } = p;
+
+			let (ids, validity_votes): (Vec<_>, Vec<_>) = validity_votes
+						.into_iter()
+						.map(|(id, vote)| (id, vote.into()))
+						.unzip();
+
+			let group = match self.table_context.groups.get(&self.assignment) {
+				Some(group) => group,
+				None => continue,
+			};
+
+			let mut validator_indices = BitVec::with_capacity(
+				group.len()
+			);
+
+			validator_indices.resize(group.len(), false);
+
+			for id in ids.iter() {
+				if let Some(position) = group.iter().position(|x| x == id) {
+					validator_indices.set(position, true);
+				}
+			}
+
+			let backed = BackedCandidate {
+				candidate,
+				validity_votes,
+				validator_indices,
+			};
+
+			res.push(NewBackedCandidate(backed.clone()));
+		}
+
+		res
+	}
+
+	/// Check if there have happened any new misbehaviors and issue necessary messages.
+	async fn issue_new_misbehaviors(&mut self) -> Result<(), Error> {
+		let mut reports = Vec::new();
+
+		for (k, v) in self.table.get_misbehavior().iter() {
+			if !self.reported_misbehavior_for.contains(k) {
+				self.reported_misbehavior_for.insert(*k);
+
+				let f = FromTableMisbehavior {
+					id: *k,
+					report: v.clone(),
+					signing_context: self.table_context.signing_context.clone(),
+					key: self.table_context.validators[*k as usize].clone(),
+				};
+
+				if let Ok(report) = MisbehaviorReport::try_from(f) {
+					let message = ProvisionerMessage::ProvisionableData(
+						ProvisionableData::MisbehaviorReport(self.parent, report)
+					);
+
+					reports.push(message);
+				}
+			}
+		}
+
+		for report in reports.drain(..) {
+			self.send_to_provisioner(report).await?
+		}
+
+		Ok(())
+	}
+
+	/// Import a statement into the statement table and return the summary of the import.
+	async fn import_statement(
+		&mut self,
+		statement: &SignedFullStatement,
+	) -> Result<Option<TableSummary>, Error> {
+		let stmt = primitive_statement_to_table(statement);
+
+		let summary = self.table.import_statement(&self.table_context, stmt);
+
+		self.issue_new_misbehaviors().await?;
+
+		return Ok(summary);
+	}
+
+	async fn process_msg(&mut self, msg: CandidateBackingMessage) -> Result<(), Error> {
+		match msg {
+			CandidateBackingMessage::Second(_, candidate, pov) => {
+				// If the message is a `CandidateBackingMessage::Second`, sign and dispatch a
+				// Seconded statement only if we have not seconded any other candidate and
+				// have not signed a Valid statement for the requested candidate.
+				match self.seconded {
+					// This job has not seconded a candidate yet.
+					None => {
+						let candidate_hash = candidate.hash();
+
+						if !self.issued_validity.contains(&candidate_hash) {
```

Comment:
A sanity check here that the candidate is within our assignment group would be helpful too, I think.